src/Utilities.ts

```diff
@@ -327,32 +308,66 @@ export class Utilities {
      * Handles cleaning up the Sarif Viewer temp directory used for temp files (embedded code, converted files, etc.)
      */
     public static removeSarifViewerTempDirectory(): void {
-        const path = Utilities.Path.join(Utilities.Os.tmpdir(), Utilities.SarifViewerTempDir);
-        Utilities.removeDirectoryContents(path);
-        Utilities.Fs.rmdirSync(path);
+        const tempPath = path.join(Utilities.Os.tmpdir(), Utilities.SarifViewerTempDir);
+        Utilities.removeDirectoryContents(tempPath);
+        fs.rmdirSync(tempPath);
+    }
+
+    /**
+     * Fixes up file schemed based paths to contain the proper casing for VSCode's URIs which are case sensitive.
+     * @param uri The URI for which to fix the casing.
+     */
+    public static fixUriCasing(uri: Uri): Uri {
+        // We can only support file-system scheme files.
+        if (uri.scheme.toUpperCase() !== "FILE") {
```

Comment:
>toUpperCase [](start = 23, length = 11)

Is there a `toUpperInvariant`? This upper-casing of the `file` scheme was the source of the notorious "Turkish i" bug.

Reply:
Couldn't hurt, though I'm not sure about adverse impact of a bad comparison. If you had files that collided in this way, we might open the wrong thing. This is no different than what would happen with any file duplication where elements differed by casing only. So, we have the same problem already (with no path to a perfect solution).

---
In reply to: [389079043](https://github.com/microsoft/sarif-vscode-extension/pull/208#discussion_r389079043) [](ancestors = 389079043)